decoder/stats.go

```diff
@@ -48,11 +48,11 @@ type shinyChecks struct {
 }
 
 type areaPokemonCountDetail struct {
-	hundos      [maxPokemonNo + 1]int
-	nundos      [maxPokemonNo + 1]int
-	shinyChecks [maxPokemonNo + 1]shinyChecks
-	count       [maxPokemonNo + 1]int
-	ivCount     [maxPokemonNo + 1]int
+	hundos      [maxPokemonNo + 1]map[int]int
```

Comment:
The reason an array was chosen (rather than a map) was because this is an extremely fast data structure and we knew the limits.  If you need a map because this is no longer possible (and I'm sure you're right, form is too sparse to hold in the same way), then you should likely move just to a map of a pokemon type that contains both pokemon id and form id.  There is no longer any advantage to holding the array if you have to go into a map immediately afterwards.

Reply:
I can convert it to array later on once I will confirm it's working as intended. I don't want to rewrite the whole structure in there, it's a mess. As for structure holding Form and Pokemon Id it would require iterating all items.

Reply:
I don’t think you can reasonably convert to an array that’s my point. A single map for pokemon/form is probably the best structure. And of course you can iterate a map by only keys present